pkg/reconciler/eventtype/eventtype.go

```diff
@@ -49,7 +54,18 @@ var _ eventtypereconciler.Interface = (*Reconciler)(nil)
 // 1. Verify the Broker/Reference exists.
 // 2. Verify the Broker/Reference is ready.
 func (r *Reconciler) ReconcileKind(ctx context.Context, et *v1beta2.EventType) pkgreconciler.Event {
-	b, err := r.getReference(et)
+	switch et.Spec.Reference.Kind {
+	case "Broker":
+		return r.reconcileBrokerReference(ctx, et)
+	case "Channel":
```

Comment:
this does not really work w/ other (3rd party) cahnnel, 

We have changed the `kresource` to be a duck:
https://github.com/knative/pkg/pull/2754

and I have a local branch, which I will PR soon, that basically does this:

```go
// ReconcileKind implements Interface.ReconcileKind.
// 1. Verify the Reference exists.
func (r *Reconciler) ReconcileKind(ctx context.Context, et *v1beta2.EventType) pkgreconciler.Event {

	_, err := r.kReferenceResolver.Resolve(ctx, et.Spec.Reference, et)
	if err != nil {
		if apierrs.IsNotFound(err) {
			logging.FromContext(ctx).Errorw("Broker reference does not exist", zap.Error(err))
			et.Status.MarkBrokerDoesNotExist()
		} else {
			logging.FromContext(ctx).Errorw("Unable to get the Broker reference", zap.Error(err))
			et.Status.MarkBrokerExistsUnknown("BrokerGetFailed", "Failed to get broker reference: %v", err)
		}
		return err
	}
	et.Status.MarkBrokerExists()

	return nil
}
```

yes: Broker needs to be renamed to `Reference`

Reply:
@matzew would we still need to track the reference, or is this no longer necessary since we only care if the reference exists?

Reply:
we track, since it can be removed after the fact

let me update my other branch / PR, now that the dep is finally in